In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/youtube-comments-dataset/YoutubeCommentsDataSet.csv


In [2]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("atifaliak/youtube-comments-dataset")

print("Path to dataset files:", path)

Path to dataset files: /kaggle/input/youtube-comments-dataset


In [3]:
import pandas as pd
data = pd.read_csv("/kaggle/input/youtube-comments-dataset/YoutubeCommentsDataSet.csv")
data.head()

Comment Sentiment
0  lets not forget that apple pay in 2014 require...   neutral
1  here in nz 50 of retailers don’t even have con...  negative
2  i will forever acknowledge this channel with t...  positive
3  whenever i go to a place that doesn’t take app...  negative
4  apple pay is so convenient secure and easy to ...  positive

In [4]:
data["Sentiment"] = data["Sentiment"].map({'neutral': 0, 'positive': 1, 'negative': -1})
data.head()

Comment  Sentiment
0  lets not forget that apple pay in 2014 require...          0
1  here in nz 50 of retailers don’t even have con...         -1
2  i will forever acknowledge this channel with t...          1
3  whenever i go to a place that doesn’t take app...         -1
4  apple pay is so convenient secure and easy to ...          1

In [5]:
import pandas as pd
data = pd.read_csv("/kaggle/input/youtube-comments-dataset/YoutubeCommentsDataSet.csv")
data.head()

Comment Sentiment
0  lets not forget that apple pay in 2014 require...   neutral
1  here in nz 50 of retailers don’t even have con...  negative
2  i will forever acknowledge this channel with t...  positive
3  whenever i go to a place that doesn’t take app...  negative
4  apple pay is so convenient secure and easy to ...  positive

In [6]:
data.columns

Index(['Comment', 'Sentiment'], dtype='object')

In [7]:
data.shape

(18408, 2)

In [8]:
data.isnull().sum()

Comment      44
Sentiment     0
dtype: int64

In [9]:
data.dropna(subset=["Comment"], inplace=True)
data.head()


Comment Sentiment
0  lets not forget that apple pay in 2014 require...   neutral
1  here in nz 50 of retailers don’t even have con...  negative
2  i will forever acknowledge this channel with t...  positive
3  whenever i go to a place that doesn’t take app...  negative
4  apple pay is so convenient secure and easy to ...  positive

In [10]:
data['Sentiment'] = data['Sentiment'].map({'positive': 1, 'negative': -1, 'neutral':0})

data.head()

Comment  Sentiment
0  lets not forget that apple pay in 2014 require...          0
1  here in nz 50 of retailers don’t even have con...         -1
2  i will forever acknowledge this channel with t...          1
3  whenever i go to a place that doesn’t take app...         -1
4  apple pay is so convenient secure and easy to ...          1

In [11]:
data.dropna(subset=["Comment"], inplace=True)


In [12]:
import pandas as pd
data = pd.read_csv("/kaggle/input/youtube-comments-dataset/YoutubeCommentsDataSet.csv")
data.head()

Comment Sentiment
0  lets not forget that apple pay in 2014 require...   neutral
1  here in nz 50 of retailers don’t even have con...  negative
2  i will forever acknowledge this channel with t...  positive
3  whenever i go to a place that doesn’t take app...  negative
4  apple pay is so convenient secure and easy to ...  positive

In [13]:
data.isnull().sum()

Comment      44
Sentiment     0
dtype: int64

In [14]:
data['Sentiment'] = data['Sentiment'].map({'positive': 1, 'negative': -1, 'neutral':0})
data.head()

Comment  Sentiment
0  lets not forget that apple pay in 2014 require...          0
1  here in nz 50 of retailers don’t even have con...         -1
2  i will forever acknowledge this channel with t...          1
3  whenever i go to a place that doesn’t take app...         -1
4  apple pay is so convenient secure and easy to ...          1

In [15]:
data.dropna(subset=["Comment"], inplace=True)
data.head()

Comment  Sentiment
0  lets not forget that apple pay in 2014 require...          0
1  here in nz 50 of retailers don’t even have con...         -1
2  i will forever acknowledge this channel with t...          1
3  whenever i go to a place that doesn’t take app...         -1
4  apple pay is so convenient secure and easy to ...          1

In [16]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 18364 entries, 0 to 18407
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Comment    18364 non-null  object
 1   Sentiment  18364 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 430.4+ KB


In [17]:
data.isnull().sum()

Comment      0
Sentiment    0
dtype: int64

In [18]:
data.columns
paras = data[["Comment", "Sentiment"]]
print(data.duplicated())


0        False
1        False
2        False
3        False
4        False
         ...  
18403    False
18404    False
18405    False
18406    False
18407    False
Length: 18364, dtype: bool


In [20]:
data["Comment_Length"] = data["Comment"].apply(len)
data.head()
    


Comment  Sentiment  \
0  lets not forget that apple pay in 2014 require...          0   
1  here in nz 50 of retailers don’t even have con...         -1   
2  i will forever acknowledge this channel with t...          1   
3  whenever i go to a place that doesn’t take app...         -1   
4  apple pay is so convenient secure and easy to ...          1   

   Comment_Length  
0             317  
1             163  
2             183  
3             450  
4             135

In [26]:
def clean_text(text):
    text = text.lower()
    text = re.sub(r"http\S+|www\S+|https\S+", '', text)
    text = re.sub(r'\W|\d', ' ', text)
    words = text.split()  # Instead of NLTK tokenizer
    return ' '.join(words)

data['Cleaned_Comment'] = data['Comment'].apply(clean_text)




In [27]:
data.head()

Comment  Sentiment  \
0  lets not forget that apple pay in 2014 require...          0   
1  here in nz 50 of retailers don’t even have con...         -1   
2  i will forever acknowledge this channel with t...          1   
3  whenever i go to a place that doesn’t take app...         -1   
4  apple pay is so convenient secure and easy to ...          1   

   Comment_Length                                    Cleaned_Comment  
0             317  lets not forget that apple pay in required a b...  
1             163  here in nz of retailers don t even have contac...  
2             183  i will forever acknowledge this channel with t...  
3             450  whenever i go to a place that doesn t take app...  
4             135  apple pay is so convenient secure and easy to ...

In [28]:
from sklearn.feature_extraction.text import TfidfVectorizer


vectorizer = TfidfVectorizer(max_features=5000)  


X = vectorizer.fit_transform(data['Cleaned_Comment'])  
y = data['Sentiment'] 


In [29]:
data.head()

Comment  Sentiment  \
0  lets not forget that apple pay in 2014 require...          0   
1  here in nz 50 of retailers don’t even have con...         -1   
2  i will forever acknowledge this channel with t...          1   
3  whenever i go to a place that doesn’t take app...         -1   
4  apple pay is so convenient secure and easy to ...          1   

   Comment_Length                                    Cleaned_Comment  
0             317  lets not forget that apple pay in required a b...  
1             163  here in nz of retailers don t even have contac...  
2             183  i will forever acknowledge this channel with t...  
3             450  whenever i go to a place that doesn t take app...  
4             135  apple pay is so convenient secure and easy to ...

In [31]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, classification_report

# Split data into training and testing sets (80% train, 20% test)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train Logistic Regression Model
model = LogisticRegression(max_iter=200)
model.fit(X_train, y_train)

# Make Predictions
y_pred = model.predict(X_test)

# Evaluate Accuracy
print("Accuracy:", accuracy_score(y_test, y_pred))
print("Classification Report:\n", classification_report(y_test, y_pred))


Accuracy: 0.7563299754968691
Classification Report:
               precision    recall  f1-score   support

          -1       0.58      0.31      0.40       441
           0       0.63      0.58      0.61       912
           1       0.81      0.91      0.86      2320

    accuracy                           0.76      3673
   macro avg       0.67      0.60      0.62      3673
weighted avg       0.74      0.76      0.74      3673



Sentiment Analysis Done with Logistic Regression Model.
The Models has achived a 75.6% Accuracy for predicting if the comment are positive,negative or neutral.
The model shows an 0.86 F1 score for predicting positive comments this means that the models performs very good at finding positive comment
The models shows an 0.61 score for predicting neutral comments this means the models performs fairly at finding neutral comments
The models shows 0.40 for predicitng negative comments this mean the models performs poorly at finding negative comments